src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -161,9 +162,23 @@ public function get(): ?array
                 ->orderBy('available_at', 'ASC')
                 ->setMaxResults(1);
 
+            // use WITH (UPDLOCK, ROWLOCK) to lock table if database is SQL Server
```

Comment:
I don't see what in the code below tell you are running `msqsl` or `sql server`

Reply:
`AbstractPlatform::appendLockHint($fromClause, $lockMode)` only returns the "WITH (UPDLOCK, ROWLOCK)" when the platform is an instance of SQLServerPlatform (this platform will also return empty string from `getWriteLockSQL()`

Reply:
Look like `SQL Anywhere` also provides such behavior.
Id rather change the comment into `append Pessimistic lock to the FROM close`